observation/point_source_scan.py

```diff
@@ -91,43 +91,43 @@ class NoTargetsUpError(Exception):
                     # Do different raster scan on strong and weak targets
                     if not opts.quick and not opts.fine:
                         if opts.source_strength == 'strong' or \
-                           (opts.source_strength == 'auto' and target.flux_density(opts.centre_freq) > 10.0):
-                            session.raster_scan(target, num_scans=5, scan_duration=30, scan_extent=6.0,
-                                                scan_spacing=0.25, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                                (opts.source_strength == 'auto' and target.flux_density(opts.centre_freq) > 10.0):
+                            raster_params = dict(num_scans=5, scan_duration=30, scan_extent=6.0, scan_spacing=0.25)
                         else:
-                            session.raster_scan(target, num_scans=5, scan_duration=60, scan_extent=4.0,
-                                                scan_spacing=0.25, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=5, scan_duration=60, scan_extent=4.0, scan_spacing=0.25)
                     else:  # The branch for Quick and Fine scans
                         if opts.quick:
-                            session.raster_scan(target, num_scans=3, scan_duration=15, scan_extent=5.0,
-                                                scan_spacing=0.5, scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=3, scan_duration=15, scan_extent=5.0, scan_spacing=0.5)
                         elif opts.fine:
-                            session.raster_scan(target, num_scans=5, scan_duration=60, scan_extent=1.0,
-                                                scan_spacing=4. / 60., scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
+                            raster_params = dict(num_scans=5, scan_duration=60, scan_extent=1.0, scan_spacing=4/60.)
                         else:  # if opts.search_fine:
-                            session.raster_scan(target, num_scans=9, scan_duration=60, scan_extent=2.0,
-                                                scan_spacing=5. / 60., scan_in_azimuth=not opts.scan_in_elevation,
-                                                projection=opts.projection)
-
+                            raster_params = dict(num_scans=9, scan_duration=60, scan_extent=2.0, scan_spacing=5/60.)
+                    
+                    # Confirm that the target will be "up" for the entire duration.
+                    raster_duration = raster_params["num_scans"] * (raster_params["scan_duration"] + 2) # Extra for slew
```

Comment:
As pointed out in Anton's comment, this code exists in CaptureSession. It also includes the `nd_time`, which is not necessarily 0. It is safer to include it as well.

Reply:
nd_time is now included in the commit - please review!